server/core/src/miscServerFunct.cpp

```diff
@@ -2983,12 +2982,24 @@ irods::error get_script_output_single_line(
     const std::string&              script_name,
     const std::vector<std::string>& args,
     std::string&                    output ) {
-    bp::ipstream output_stream;
-    bp::child c(script_language, args, bp::std_out > output_stream);
-    output.clear();
-    std::getline(output_stream, output);
-    c.wait();
-    return SUCCESS();
+    namespace bp = boost::process;
+    try {
+        auto tmp = args;
+        tmp.insert(tmp.begin(), fmt::format("{}/scripts/{}", irods::get_irods_home_directory().string(), script_name));
+        bp::ipstream output_stream;
+        bp::child c(script_language, tmp, bp::std_out > output_stream);
+        output.clear();
+        if (!output_stream) {
+            return ERROR(FILE_READ_ERR, "Bad stream");
+        }
+        std::getline(output_stream, output);
+        c.wait();
+        return SUCCESS();
```

Comment:
We need to consider the exit code of the script and what that means to this function.
```cpp
const int ec = c.exit_code();
```

Reply:
Well, I'm not sure what it should do in the new function, but in the old implementation it was checked for an error closing the stream, but otherwise discarded.

Reply:
That doesn't sound like the right approach. A bad exit code indicates we need to check things out.

Again, this is something that requires discussion as a group.